common/serde-reflection/src/format.rs

```diff
@@ -167,45 +241,75 @@ impl FormatHolder for VariantFormat {
         }
         Ok(())
     }
-
-    fn normalize(&mut self) -> Result<()> {
-        match self {
-            Self::Unknown => Err(Error::UnknownFormat),
-            Self::Unit => Ok(()),
-            Self::NewType(format) => format.normalize(),
-            Self::Tuple(formats) => {
-                for format in formats {
-                    format.normalize()?;
-                }
-                Ok(())
-            }
-            Self::Struct(named_formats) => {
-                for format in named_formats {
-                    format.normalize()?;
-                }
-                Ok(())
-            }
-        }
-    }
 }
 
 impl<T> FormatHolder for Named<T>
 where
     T: FormatHolder + std::fmt::Debug,
 {
+    fn visit(&self, f: &mut dyn FnMut(&Format) -> Result<()>) -> Result<()> {
```

Comment:
Having the closure being taken in as an `&mut` was a little weird to see but I do understand why you're doing it that way.